tft_bot/helpers/click_helpers.py

```diff
@@ -57,142 +48,24 @@ def click_right(delay=0.1) -> None:
     move_duration: float = random.uniform(0.1, 1.0),
     delay: float = 0.2,
-    offset: float | str = "half",
     action: str = "left",
-) -> bool:
+) -> None:
     """Attempt to click to the (relative) middle of the specified image.
 
     Args:
-        path (str): The relative or absolute path to the image to be clicked.
-        precision (float, optional): The precision to be used when matching the image.
-            Defaults to 0.8.
+        image_details: The coordinates to click to, and the height and width
         move_duration (float, optional): Time taken for the mouse to move.
             Defaults to random.uniform(0.1, 1.0).
         delay (float, optional): The delay between mouse down & up. Defaults to 0.2.
-        offset (float | str, optional): When specified, the manual offset from the relative center in pixels.
-            Defaults to "half".
-        action (str, optional): The mouse button to perform. Defaults to "left".
-
-    Returns:
-        bool: True if successfully clicked, False otherwise.
-    """
-    path = resource_path(path)
-    pos = find_image(path, precision)
-    if pos:
-        try:
-            return click_image_rand(
-                path,
-                pos,
-                action,
-                move_duration=move_duration,
-                delay=delay,
-                offset=offset,
-            )
-        except Exception as err:
-            logger.debug(f"M|Failed to click to {err}")
-    else:
-        logger.debug(f"M|Could not find '{path}', skipping")
-    return False
-
-
-def click_to_middle_multiple(
-    images: list[str],
-    precision: float = 0.8,
-    conditional_func: Callable | None = None,
-    delay: float = None,
-    action: str = "left",
-) -> bool:
-    """Attempt to click to the (relative) middle of the specified images.
-
-    Args:
-        images (list[str]): The list of relative or absolute paths to images to attempt clicking.
-        precision (float, optional): The precision to be used when matching the image. Defaults to 0.8.
-        conditional_func (Callable | None, optional): The function to evaluate if the click was successful.
-            Defaults to None.
-        delay (float, optional): The delay between mouse down & up. Defaults to None.
         action (str, optional): The mouse button to perform. Defaults to "left".
-
-    Returns:
-        bool: _description_
     """
-    for image in images:
-        image = resource_path(image)
-        try:
-            click_to_middle(image, precision=precision, action=action)
-        except Exception:
-            logger.debug(f"M|Failed to click {image}")
-        if generic_helpers.is_var_number(delay):
-            time.sleep(delay)
-        if generic_helpers.is_var_function(conditional_func) and conditional_func():
-            return True
-    return False
+    auto.moveTo(
+        image_details[0] + image_details[2] / 2,
+        image_details[1] + image_details[3] / 2,
```

Comment:
I do completely understand this (after re-reading it a few times), but I wonder if either of these approaches may make sense:
- Refactor `image_details` to instead split up into two params:
  - `image_coordinates`, containing X & Y coords only, or X1,Y1,X2,Y2 (I'm fine with either approach, we could even make those separate "classes"
  - `image_details`, containing height & width (and other future image-specific params we may add, though I can't think of any)
  - This could be done via use of the `NamedTuple` approach (or something else, this was just the first thing to come to mind)
    - https://stackoverflow.com/a/53410614/4654928
    - Has the benefit of being "typed" in that sense, plus seems to be what most of the community is rallying behind to ensure 'strict' usage
  - Has the benefit of being much more readable than `tuple[int, int, int, int]` 😅  


A second point that I am not sure if it makes sense to switch out into a second comment, I think if we go the approach I suggested, condensing to one `click_to` function may make sense, and it could either be a boolean which defaults true/false for if to click to middle, and if so it would try to utilize the `image_details` that were passed.

Not deadset on that as I do like specific functions, they just seem very duplicated and it'd only be a one or two line difference, I think? 🤔

Reply:
Yeah, namedtuple works, I've also been thinking of making it a quick dataclass. I already planned to go through the methods and clean them up even more, so this is very much appreciated feedback (for both points)! For now, I've been mainly moving around existing logic.